File: Features/confidentialidentity-whistleblower/workflows/src/test/java/net/corda/samples/whistleblower/flows/FlowTests.java
package net.corda.samples.whistleblower.flows;

import com.google.common.collect.ImmutableList;
import net.corda.core.concurrent.CordaFuture;
import net.corda.core.identity.CordaX500Name;
import net.corda.core.transactions.SignedTransaction;
import net.corda.testing.node.*;
import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import org.junit.Assert;

import java.util.concurrent.ExecutionException;

public class FlowTests {
    private final MockNetwork network = new MockNetwork(new MockNetworkParameters()
            .withCordappsForAllNodes(ImmutableList.of(
                    TestCordapp.findCordapp("net.corda.samples.whistleblower.contracts"),
                    TestCordapp.findCordapp("net.corda.samples.whistleblower.flows")))
            .withNotarySpecs(ImmutableList.of(new MockNetworkNotarySpec(CordaX500Name.parse("O=Notary,L=Nakuru,C=KE")))));
    private final StartedMockNode a = network.createNode();
    private final StartedMockNode b = network.createNode();
    private final StartedMockNode c = network.createNode();

    public FlowTests() {
        ImmutableList.of(a, b).forEach(it -> it.registerInitiatedFlow(BlowWhistleFlowResponder.class));
    }

    @Before
    public void setUp() throws Exception {
        network.startNodes();
    }

    @After
    public void tearDown() {
        network.stopNodes();
    }

    //simple unit test to check the public keys that used in the transaction
    //are different from both mock node a's legal public key and mock node b's legal public key.
    @Test
    public void dummyTest() throws ExecutionException, InterruptedException {
        CordaFuture<SignedTransaction> future = a.startFlow(new BlowWhistleFlow(b.getInfo().getLegalIdentities().get(0), c.getInfo().getLegalIdentities().get(0)));
        network.runNetwork();
        SignedTransaction ptx = future.get();
        Assert.assertTrue(!ptx.getTx().getRequiredSigningKeys().contains(a.getInfo().getLegalIdentities().get(0).getOwningKey()));
        Assert.assertTrue(!ptx.getTx().getRequiredSigningKeys().contains(b.getInfo().getLegalIdentities().get(0).getOwningKey()));
    }
}

File: kotlin-samples/Features/confidentialIdentity-whistleblower/workflows/src/test/kotlin/net/corda/samples/whistleblower/flows/FlowTests.kt
package net.corda.samples.whistleblower.flows

import net.corda.core.identity.CordaX500Name
import net.corda.testing.node.MockNetwork
import net.corda.testing.node.MockNetworkNotarySpec
import net.corda.testing.node.MockNetworkParameters
import net.corda.testing.node.TestCordapp
import org.junit.After
import org.junit.Before
import org.junit.Test

class FlowTests {
    private val network = MockNetwork(MockNetworkParameters(cordappsForAllNodes = listOf(
            TestCordapp.findCordapp("net.corda.samples.whistleblower.contracts"),
            TestCordapp.findCordapp("net.corda.samples.whistleblower.flows")
    ),
            notarySpecs = listOf(MockNetworkNotarySpec(CordaX500Name("Notary","Nakuru","KE")))
    ))
    private val a = network.createNode()
    private val b = network.createNode()
    private val c = network.createNode()

    init {
        listOf(a, b).forEach {
            it.registerInitiatedFlow(BlowWhistleFlowResponder::class.java)
        }
    }

    @Before
    fun setup() = network.runNetwork()

    @After
    fun tearDown() = network.stopNodes()

    //simple unit test to check the public keys that used in the transaction
    //are different from both mock node a's legal public key and mock node b's legal public key.
    @Test
    fun `dummy test`() {
        val future = a.startFlow(BlowWhistleFlow(b.info.legalIdentities.first(), c.info.legalIdentities.first()))
        network.runNetwork()
        val ptx = future.get()
        assert(!ptx.tx.requiredSigningKeys.contains(a.info.legalIdentities.first().owningKey))
        assert(!ptx.tx.requiredSigningKeys.contains(b.info.legalIdentities.first().owningKey))

    }
}
